tmpl/wiki/deployment.md

```diff
@@ -146,64 +161,90 @@ a small `travis-senv` command-line tool to make this easier.
+       \ git commit -m \"adding $TRAVIS_COMMIT for $MIRAGE_BACKEND\" &&\
+       \ git push"
+end
 ```
 
-While this fragment is a bit longer than the rest, it's all rather
-straightforward.  We install `travis-senv` and use it to retrieve
-our SSH deployment key (which is stored encrypted in the `.travis.yml`
-file, and can only be decrypted by the Travis infrastructure).
-Once that's installed, we initialize `git` and clone the deployment
-repository.  We then compress our output kernel and commit it to
-the repository along with the configuration data.
+(The `?|` and `?|~` operators are defined in [`yorick.ml`](https://github.com/ocaml/ocaml-travisci-skeleton/blob/master/yorick.ml) and act simply as shell
+invocations piping `stdin`/`stdout`/`stderr` in the usual way, with
+success/failure tests on the status of each invoked command.)
+
+While this fragment is a bit longer than the rest, it's actually rather
+straightforward. We install `travis-senv` and use it to retrieve our SSH
+deployment key (which is stored encrypted in the `.travis.yml` file, and can
+only be decrypted by the Travis infrastructure). Once that's installed, we
+initialise `git` and clone the deployment repository. We then compress our
+output kernel, update the `xen/latest` pointer to refer to the new kernel,
+commit everything, and push the results back.
 
 ### Updating the live website
 
-The live website runs on a Xen machine hosted in Cambridge, and
-this simply polls the [mirage/mirage-www-deployment](http://github.com/mirage/mirage-www-deployment)
-repository for new pushes, and updates to the latest version in
-that repository when one appears.  This machine doesn't need to
-have any of the MirageOS tools installed, and if there's an error in
-an updated website, we can simply revert back to a previous changeset
-of the deployed site.
+The live website runs on a Xen machine which has a crontab entry in dom0 that
+polls the
+[mirage/mirage-www-deployment](http://github.com/mirage/mirage-www-deployment)
+repository for new content via `git pull`. When new content __is__ obtained, a
+Git [`merge-hook`](https://github.com/mirage/mirage-www-deployment) is invoked
+which stops the current unikernel and boots the newly updated latest version.
+This machine doesn't need to have any of the MirageOS tools installed, and if
+there's an error in an updated website, we can simply revert back to a previous
+as previous binary revisions of the site remain in the deployment repository.
 
 The same approach could be used to deploy to Heroku, Rackspace or EC2 instead of
 a custom Xen host; we'll cover this in future updates.
 
-*(Note: This article is based on a similar blog post for general OCaml use available [here](http://anil.recoil.org/2013/10/06/travis-secure-ssh-integration.html))*
+*(Note: This article is based on a similar blog post for general OCaml use
+ available
+ [here](http://anil.recoil.org/2013/10/06/travis-secure-ssh-integration.html).)*
```

Comment:
The following use specific examples to guide the user through details of what the scripts are doing — though the scripts themselves re general-purpose. .

- http://amirchaudhry.com/from-jekyll-to-unikernel-in-fifty-lines/
- http://amirchaudhry.com/heroku-for-unikernels-pt1/